pkg/controller/utils/secrets.go

```diff
@@ -2,23 +2,76 @@ package utils
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
+	"github.com/openshift/hive/pkg/constants"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/clientcmd/api"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// LoadSecretData loads a given secret key and returns it's data as a string.
+// LoadSecretData loads a given secret key and returns its data as a string.
 func LoadSecretData(c client.Client, secretName, namespace, dataKey string) (string, error) {
 	s := &corev1.Secret{}
 	err := c.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: namespace}, s)
 	if err != nil {
 		return "", err
 	}
-	retStr, ok := s.Data[dataKey]
+	retBytes, ok := s.Data[dataKey]
 	if !ok {
 		return "", fmt.Errorf("secret %s did not contain key %s", secretName, dataKey)
 	}
-	return string(retStr), nil
+	switch dataKey {
+	case constants.KubeconfigSecretKey, constants.RawKubeconfigSecretKey:
+		if _, err := validateKubeconfig(retBytes); err != nil {
+			return "", err
+		}
+	}
+	return string(retBytes), nil
+}
+
+// RestConfigFromSecret accepts a Secret containing `kubeconfig` and/or `raw-kubeconfig` keys
+// and returns a rest.Config loaded therefrom.
+// If tryRaw is true, we will look for `raw-kubeconfig` first and use it if present, falling
+// back to `kubeconfig` otherwise.
+// The error return is non-nil if:
```

Comment:
```suggestion
// The error returned is non-nil if:
```

Reply:
This was intentionally a noun ("the error return variable"?) but I can change it in a fup if it bugs ya :)

Reply:
Eh - either makes sense to me so ignore